libs/src/utils/promiseFight.ts

```diff
@@ -4,10 +4,12 @@
  * @param captureErrored optional capture errored promises
  * @return A promise that resolves with the first promise that resolves
  */
-export async function promiseFight (promises: Array<Promise<unknown>>, captureErrored = false) {
-  const errored: unknown[] = []
-  return await Promise.all(promises.map(async p => {
-    return await p.then(
+export async function promiseFight<T, T2> (promises: Array<Promise<T>>): Promise<T | T2>
```

Comment:
what's `T` and `T2` here? I want to make sure I'm understanding this correctly

Reply:
yeah the types could be better named (going to rename T to T1), but they come from the definition of Promise so kept them the same name. So T1 is the type passed to the resolve function of the promise, and T2 is the type passed to the reject function of the promise. Long story short, what ive gathered with `promiseFight` is that it returns either `val` ie the expected resolve type for the promises, or {val, errored}, which is kinda like the resolved and rejected type together. super weird function to type ultimately!

Reply:
thanks for the context here!